### ISSUE TITLE
Method 'get_Info' in type 'MySql.Data.EntityFrameworkCore.Infraestructure.MySQLOptionsExtension' from assembly 'MySql.Data.EntityFrameworkCore'  does not have an implementation 

### ISSUE THREAD

**Issue by khteh** [dotnet/efcore#17788]
I have upgraded the SDK and runtime to 3.0, upgraded the packages in .csproj but bump into the exception:
```
System.TypeLoadException: Method 'get_Info' in type 'MySql.Data.EntityFrameworkCore.Infraestructure.MySQLOptionsExtension' from assembly 'MySql.Data.EntityFrameworkCore, Version=8.0.17.0, Culture=neutral, PublicKeyToken=c5687fc88969c44d' does not have an implementation.
```
Notice the spelling error in the namespace, `Infraestructure` instead of `Infrastructure`. Could that be the root cause?

### Steps to reproduce

```
$ dn --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100
 Commit:    04339c3a26

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  19.04
 OS Platform: Linux
 RID:         ubuntu.19.04-x64
 Base Path:   /usr/share/dotnet-3.0.100/sdk/3.0.100/

Host (useful for support):
  Version: 3.0.0
  Commit:  7d57652f33

.NET Core SDKs installed:
  3.0.100 [/usr/share/dotnet-3.0.100/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0 [/usr/share/dotnet-3.0.100/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0 [/usr/share/dotnet-3.0.100/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

### Further technical details

EF Core version: 3.0
Database provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer) `MySql.Data.EntityFrameworkCore`
Target framework: .NET Core 3.0
Operating system: Ubuntu 19.04
IDE: Visual Studio Code 1.38.1

**Comment by smitpatel** [comment]
You would need to wait until the package author publishes a version which is compatible with EF Core 3.0. EF Core 3 is major release which contains breaking changes and each database provider need to react to it. Closing as this is external issue. Please contact the author of the package you are trying to use.

**Comment by khteh** [comment]
I tried downgrade `Microsoft.EntityFrameworkCore` to 2.2.6 but the exception persists.

**Comment by smitpatel** [comment]
Share your csproj file. You need to use versions for 2.2 for all asp.net packages. Just downgrading EFCore package will not get you anything.

**Comment by khteh** [comment]
Forget it. Too much unnecessary work. Either wait for donkey years before Oracle works it out or switch to https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/797

**Comment by ifew** [comment]
We should downgrade from EFCore 3.0 to 2.2 ??

**Comment by smitpatel** [comment]
@ifew - You will need to wait for MySQL provider which works with EF Core 3.0. Till that is released, you would need to stick to version 2.2

**Comment by hamid-unifi** [comment]
has been reported here as well:
https://bugs.mysql.com/bug.php?id=97000

**Comment by j0nimost** [comment]
THIS SUCKS!!!!

**Comment by roji** [comment]
Note that version 3.0 of the Pomelo MySQL provider [has already been released](https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql), so this problem should no longer occur. If anyone is still running into issues, please open a new issue on the [Pomelo repository](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql).

**Comment by yasuoyuhao** [comment]
MySql.Data.EntityFrameworkCore release new version 8.0.19, The situation has not changed.

**Comment by roji** [comment]
@yasuoyuhao and others, just to clarify the current situation... There are two MySQL providers for Entity Framework Core:

* The official one from MySQL: [MySql.Data.EntityFrameworkCore](https://www.nuget.org/packages/MySql.Data.EntityFrameworkCore). As of now, the latest version is 8.0.19, and works with Entity Framework Core 2.1 (and probably also 2.2). Since EF Core 3.0 is a major version with breaking changes, you cannot use it with this provider.
* The Pomelo provider: [Pomelo.EntityFrameworkCore.MySql](https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql). There is a 3.1 version of this provider.

In other words, if you want to use EF Core 3.0/3.1 with MySQL, at this point you need to use the Pomelo provider (or wait for the official MySQL one to get released).

Note that neither of these are provided by the EF team or tracked by this repo - any bugs or feature requests must be filed directly with its respective project.

**Comment by rsmastersoft95** [comment]
try out these steps:

1. you need to remove all entity framework packages ( ex: EntityFrameworkCore.SqlServer or EntityFramework.Tools ).
2. Now Reinstall MySql.EntityFramework.Core

**Comment by scubakiz** [comment]
This is listed as the official package on Microsoft's site:

https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql

**Comment by ajcvickers** [comment]
@scubakiz Can you point to where this is linked as an "official package"?

**Comment by scubakiz** [comment]
This is Microsoft's list of Database Providers for .Net Core:

https://docs.microsoft.com/en-us/ef/core/providers/?tabs=dotnet-core-cli

There are multiple options for MySQL available.

**Comment by ajcvickers** [comment]
@scubakiz That doesn't indicate in any way that those packages are "official". They are just the packages that people have told us about. Hence:

> EF Core providers are built by a variety of sources. Not all providers are maintained as part of the Entity Framework Core Project. When considering a provider, be sure to evaluate quality, licensing, support, etc. to ensure they meet your requirements. Also make sure you review each provider's documentation for detailed version compatibility information.

That being said, the Pomelo MySQL provider seems very good to me. It's just not officially endorsed in any way by Microsoft.

**Comment by karan12990** [comment]
Same error.  

`Method 'get_Info' in type 'MySql.Data.EntityFrameworkCore.Infraestructure.MySQLOptionsExtension' from assembly 'MySql.Data.EntityFrameworkCore, Version=8.0.19.0, Culture=neutral, PublicKeyToken=c5687fc88969c44d' does not have an implementation`

Still no updated version for 3.0.

**Comment by CarlosLanderas** [comment]
Suffering the same problem

**Comment by roji** [comment]
For anybody coming to this issue, please read [this comment](https://github.com/dotnet/efcore/issues/17788#issuecomment-574569529) above explaining the situation.

Long story short, the EF Core team is not the maintainer of MySql.Data.EntityFrameworkCore - please communicate with them about releasing a newer version of their provider which supports EF Core 3.1. As an alternative, consider switching to [Pomelo.EntityFrameworkCore.MySql](https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql) which is a full-featured EF Core MySQL provider supporting 3.1 (also not maintained by this team).

Posting further comments here won't help the situation in any way...

**Comment by CarlosLanderas** [comment]
Thanks for the information @roji

**Comment by sawankumarbundelkhandi** [comment]
This is the screenshot from the bug - https://bugs.mysql.com/bug.php?id=96990

![image](https://user-images.githubusercontent.com/10863510/81078573-100eb400-8f0c-11ea-9b8c-78a7812a5f7b.png)

And MySql.Data 8.0.20 is released on 27 April 2020

https://www.nuget.org/packages/MySql.Data/8.0.20
